### DISCUSSION TITLE
No message provided in seq logging UI

### DISCUSSION THREAD

**Discussion by lukeymyuan** [datalust/seq-tickets#1561] [Help]
Hi team, 

I have installed seq & seq pino, looks like logs have been streamed to seq, however in the UI the message does not seems to be popped up or can be filtered 

can someone help me? is it a seq configuration issue or pino issue

<img width="1245" alt="Screen Shot 2022-04-28 at 11 51 11 am" src="https://user-images.githubusercontent.com/22341913/165660312-bcde6d14-143f-4cc1-a7d3-b0037890a2eb.png">

**Comment by liammclennan** [comment]
@lukeymyuan I suspect this is because the message property is not being sent. Can you please provide some information about how you are using pino-seq and how you are calling its API? 

Either here or email to `support@datalust.co`.

**Reply by lukeymyuan** [reply]
@liammclennan  thanks for the quick response

I am using https://github.com/datalust/pino-seq 
Out-of-process (transport) usage, basically i just piped pino-seq

in terms of i use it, I have seq as logger and did sth like 

`this._logger.info(
          { type, event: params?.event },
          'Sent event to Segment'
        );`

**Reply by lukeymyuan** [reply]
version

"pino": "^6.7.0",
"pino-seq": "^0.9.0",

**Comment by liammclennan** [comment]
Hi @lukeymyuan 

I've created the following sample:

```js
const logger = require('pino')()
logger.info( { type: "crash", event: { path: 'a', count: 10 } }, 'Sent event to Segment' );
```

I run it with the out-of-process usage:

```shell
> node index.js | pino-seq --serverUrl http://localhost
```

and the result in Seq is:

![image](https://user-images.githubusercontent.com/271514/165683001-31621e04-c5fe-4376-bc07-96e82023620c.png)

Try running your app without piping to pino-seq. You should get a json object with a `msg` property. If the `msq` property has a value this is what you will see in Seq.

**Reply by lukeymyuan** [reply]
> Hi @lukeymyuan
> 
> I've created the following sample:
> 
> ```js
> const logger = require('pino')()
> logger.info( { type: "crash", event: { path: 'a', count: 10 } }, 'Sent event to Segment' );
> ```
> 
> I run it with the out-of-process usage:
> 
> ```shell
> > node index.js | pino-seq --serverUrl http://localhost
> ```
> 
> and the result in Seq is:
> 
> ![image](https://user-images.githubusercontent.com/271514/165683001-31621e04-c5fe-4376-bc07-96e82023620c.png)
> 
> Try running your app without piping to pino-seq. You should get a json object with a `msg` property. If the `msq` property has a value this is what you will see in Seq.

Hi @liammclennan 
looks like i have _message_ instead of _msg_

`message: "Sent event to Segment"
[app] [1651188525741] INFO       (37731 on Lukes-MacBook-Pro.local):
[app]     environment: "development"`

could this be the issue? and is there any workaround?

**Reply by liammclennan** [reply]
The default pino configuration will work. From the pino documentation it looks like you may have configured Pino to use a different [message key](https://github.com/pinojs/pino/blob/master/docs/api.md#messagekey-string).

**Reply by lukeymyuan** [reply]
@liammclennan yes just found out we had _messageKey: 'message'_
could there be any workaround for seq to capture "message" instead?

**Reply by liammclennan** [reply]
The easiest way is to let Pino output `msg`. If you don't want to do that you could use [`jq`](https://stedolan.github.io/jq/) to transform the output:

`node index.js | jq '.["msg"] = .message | del(.message)'`

or to leave `msg` and `message`:

`node index.js | jq '.["msg"] = .message'`

**Comment by lukeymyuan** [comment]
@liammclennan 
on the other hand, message under error object can be captured (shown in UI) for some reason

`error: {
[admin]       "type": "Error",
[admin]       "message": "Missing signature",
[admin]       "stack":
[admin]           Error: Missing signature`

**Reply by liammclennan** [reply]
`msg` is a special key that Seq maps to its `@Message` property that is displayed in the Seq UI.

**Reply by lukeymyuan** [reply]
@liammclennan 

workaround?
